format/partition-statistics-spec.md

```diff
@@ -0,0 +1,43 @@
+---
+title: "Partition Statistics Spec"
+url: partition-statistics-spec
+toc: true
+disableSidebar: true
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Partition Statistics file format
+
+This is a specification for partition statistics files. It is designed to store statistics information 
+for every partition value as a row in the **table default format** sorted based on the `partition`.
+
+The schema of the file is as follows:
+
+| Field Id | Field Name              | Field Type   | isOptional | Doc                                                                                                                         |
+|----------|-------------------------|--------------|------------|-----------------------------------------------------------------------------------------------------------------------------|
+| 1        | partition               | StructType   | false      | See [PartitionData](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/PartitionData.java) |                                                                                  
+| 2        | spec_id                 | IntegerType  | false      | Partition spec id                                                                                                           |
+| 3        | data_record_count       | LongType     | false      | Count of records in data files                                                                                              |                                               
+| 4        | data_file_count         | IntegerType  | false      | Count of data files                                                                                                         |
```

Comment:
Are we going to add file size? Although it is highly related to the record count, it is still quite useful to help understand the storage size of a partition. Besides, it is easy to get, a command like this will work.
```
SELECT sum(file_size_in_bytes) FROM db.table.files GROUP By partition
```

Reply:
+1, also for delete files

Reply:
There was also an ask for the `latest sequence number` per partition, the `last modified time` per partition along with `file size`. 
Also, I remember some ask for lower bounds and upper bounds for other columns per partition. 

These are not must-haves (but good to have). So, the plan is to brainstorm on these in phase 2 implementation as mentioned in the design doc/iceberg sync. 

>  SELECT sum(file_size_in_bytes) FROM db.table.files GROUP By partition

the plan is to incrementally compute these per snapshot. But yeah, manifest's data file entry already has this and we just need to sum this. Let us discuss more on this in phase 2.

Reply:
added `file_size_in_bytes`